backend/layers/processing/process_download.py

```diff
@@ -75,37 +91,108 @@ def remove_prefix(string: str, prefix: str) -> str:
+        else:
+            prefix = f"{os.environ['DEPLOYMENT_STAGE']}"
+        job_definition_name = f"dp-{prefix}-ingest-process-{dataset_version_id}"
+        return job_definition_name
+
+    @staticmethod
+    def estimate_resource_requirements(
+        adata: scanpy.AnnData,
+        memory_modifier: float = MEMORY_MODIFIER,
+        min_memory_MB: int = MIN_MEMORY_MB,
+        min_vcpu: int = MIN_VCPU,
+        max_memory_MB: int = MAX_MEMORY_MB,
+        max_vcpu: int = MAX_VCPU,
+        swap_memory_MB: int = SWAP_MEMORY_MB,
+        memory_per_vcpu: int = MEMORY_PER_VCPU,
+    ) -> Dict[str, int]:
+        """
+        Estimate the resource requirements for a given dataset
+
+        :param adata: The datasets AnnData object
+        :param memory_modifier: A multiplier to increase/decrease the memory requirements by
+        :param min_memory_MB: The minimum amount of memory to allocate.
+        :param min_vcpu: The minimum number of vCPUs to allocate.
+        :param max_memory_MB: The maximum amount of memory to allocate.
+        :param max_vcpu: The maximum number of vCPUs to allocate.
+        :param memory_per_vcpu: The amount of memory to allocate per vCPU.
+        :param swap_memory_MB:
+        :return: A dictionary containing the resource requirements
+        """
+        # Note: this is a rough estimate of the uncompressed size of the dataset. This method avoid loading the entire
+        # dataset into memory.
+        uncompressed_size_MB = adata.n_obs * adata.n_vars / MB
+        estimated_memory_MB = max([int(ceil(uncompressed_size_MB * memory_modifier)), min_memory_MB])
+        if estimated_memory_MB > max_memory_MB:
+            estimated_memory_MB = max_memory_MB
+            estimated_vcpus = max_vcpu
+            max_swap = swap_memory_MB
+        else:
+            estimated_vcpus = max([int(ceil(estimated_memory_MB / memory_per_vcpu)), min_vcpu])
+            max_swap = 0
+
+        return {"Vcpus": estimated_vcpus, "Memory": estimated_memory_MB, "MaxSwap": max_swap}
+
+    def create_batch_job_definition_parameters(self, local_filename: str, dataset_version_id: str) -> Dict[str, Any]:
+        adata = scanpy.read_h5ad(local_filename, backed="r")
+        batch_resources = self.estimate_resource_requirements(adata)
+        job_definition_name = self.get_job_definion_name(dataset_version_id)
+
+        return {  # Using PascalCase to match the Batch API
+            "JobDefinitionName": job_definition_name,
+            "Vcpus": batch_resources["Vcpus"],
+            "Memory": batch_resources["Memory"],
+            "LinuxParameters": {
+                "Swappiness": 60,
+                "MaxSwap": batch_resources["MaxSwap"],
+            },
+        }
+
+    def process(
+        self, dataset_version_id: DatasetVersionId, dataset_uri: str, artifact_bucket: str, sfn_task_token: str
+    ):
         """
         1. Download the original dataset
         2. Upload the labeled dataset to the artifact bucket
```

Comment:
remove step 2 from the docstring, this is now done in process_validate. move to process_validate.process if not already there